src/delay.rs

```diff
@@ -40,89 +42,56 @@ impl Delay {
     pub fn reset(&mut self, deadline: Instant) {
         self.deadline = deadline;
         self.initialized = false;
-        if let Some(task) = &self.task {
-            task.notify()
-        }
     }
 }
 
 impl Future for Delay {
-    type Item = ();
-    type Error = IoError;
+    type Output = Result<(), IoError>;
 
-    fn poll(&mut self) -> Result<Async<Self::Item>, Self::Error> {
+    fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
         if !self.initialized {
             let now = Instant::now();
             let duration = if self.deadline > now {
                 self.deadline - now
             } else {
-                return Ok(Async::Ready(()));
+                return Poll::Ready(Ok(()));
             };
             self.timerfd
                 .set_state(TimerState::Oneshot(duration), SetTimeFlags::Default);
             self.initialized = true;
         }
-        try_ready!(self.timerfd.poll_read());
-        Ok(Async::Ready(()))
+        let mut buf = [0u8; 8];
+        let mut buf = ReadBuf::new(&mut buf);
+        let timer = &mut self.as_mut().timerfd;
+        tokio::pin!(timer);
+        ready!(timer.poll_read(cx, &mut buf)?);
```

Comment:
When the type is `Unpin`, it is more idiomatic to use `Pin::new` rather than the `tokio::pin!` macro.
```suggestion
        ready!(Pin::new(&mut self.as_mut().timerfd).poll_read(cx, &mut buf)?);
```

Reply:
Thanks! I applied it here and in interval as well.

Reply:
Also, as I see you're a tokio maintainer, do you mind looking at [this loop](https://github.com/polachok/tokio-timerfd/pull/3/commits/5cef258df2dae9c9e2aa73c7a4221307a4588dd7#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R78). I'm not sure if I'm using `try_io()` correctly as the docs aren't very clear on its usage.

Reply:
I had a very quick look now, and it seems alright. The best place in the docs to look are in the examples on the `AsyncFd` struct itself. I will give a more thorough review later.

Reply:
I had another look, and this part is fine. :+1: